crash_similarity.py

```diff
@@ -50,10 +50,16 @@ def read_corpus(fnames):
         processed = preprocess(proto_signature)
 
         if frozenset(processed) not in already_selected:
-            elems.append((processed, data['signature']))
-        already_selected.add(frozenset(processed))
+            if embedding_algo == 'doc2vec':
```

Comment:
IMO it's better to add abstraction level here to make code more readable and easier to extend.
It can be map from algorithm name to input process function or base class `Algo` with methods like `prepare_corpus` and `train` extended by every new algorithm